src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +109,845 @@ public static void init()
+            loadSchematicsForPrefix(cacheSchematicFolder.toPath(), SCHEMATICS_CACHE);
+        }
 
-            if (!decorationFolder.exists() && !decorationFolder.mkdirs())
-            {
-                Log.getLogger().warn("Failed to create directories for dynamic decorations.");
-            }
-            loadStyleMaps(decorationFolder.toPath());
+        if (md5Map.size() == 0)
+        {
+            Log.getLogger().error("Error loading StructureProxy directory. Things will break!");
         }
-        catch (@NotNull IOException | URISyntaxException e)
+    }
+
+    /**
+     * Load all schematics from the scan folder.
+     */
+    @SideOnly(Side.CLIENT)
+    public static void loadScannedStyleMaps()
+    {
+        if (!allowPlayerSchematics && FMLCommonHandler.instance().getMinecraftServerInstance() == null)
         {
-            Log.getLogger().error("Error loading StructureProxy directory. Things will break!", e);
+            return;
         }
+
+
+        schematicsMap.remove(SCHEMATICS_SCAN);
+        final File schematicsFolder = Structure.getClientSchematicsFolder();
+        checkDirectory(schematicsFolder.toPath().resolve(SCHEMATICS_SCAN).toFile());
+        loadSchematicsForPrefix(schematicsFolder.toPath(), SCHEMATICS_SCAN);
     }
 
     /**
      * Load all style maps from a certain path.
+     * load all the schematics inside the folder path/prefix
+     * and add them in the md5Map
      *
      * @param basePath the base path.
-     * @throws IOException if nothing found.
+     * @param prefix   either schematics, scans, cache
      */
-    public static void loadStyleMaps(final Path basePath) throws IOException
+    private static void loadSchematicsForPrefix(@NotNull final Path basePath, @NotNull final String prefix)
     {
-        try (Stream<Path> walk = Files.walk(basePath))
+        Log.getLogger().info("Structures: loadSchematicsForPrefix(" + basePath + ", " + prefix + ")");
+        try (Stream<Path> walk = Files.walk(basePath.resolve(prefix)))
         {
             final Iterator<Path> it = walk.iterator();
 
             while (it.hasNext())
             {
                 final Path path = it.next();
 
-                final String style = path.getParent().getFileName().toString();
-
-                //Don't treat generic schematics as decorations or huts - ex: supply ship
-                if (NULL_STYLE.equals(style) || MINER_STYLE.equals(style))
-                {
-                    continue;
-                }
-
                 if (path.toString().endsWith(SCHEMATIC_EXTENSION))
                 {
-                    final String filename = path.getFileName().toString().split("\\.nbt")[0];
-                    final String hut = filename.split("\\d+")[0];
+                    String relativePath = path.toString().substring(basePath.toString().length()).split("\\" + SCHEMATIC_EXTENSION)[0];
+                    if (!path.getFileSystem().getSeparator().equals("/"))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "/" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1132)